rcl/test/rcl/test_wait.cpp

```diff
@@ -269,15 +273,16 @@ TEST(CLASSNAME(WaitSetTestFixture, RMW_IMPLEMENTATION), multi_wait_set_threaded)
   //   ss << "]";
   //   printf("%s\n", ss.str().c_str());
   // };
+
   size_t loop_count = 0;
   while (loop_test()) {
     loop_count++;
-    // print_state("triggering, current states: ");
     for (const auto & test_set : test_sets) {
       ret = rcl_trigger_guard_condition(&test_set.guard_condition);
       EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string_safe();
     }
-    std::this_thread::sleep_for(trigger_period);
+    std::this_thread::yield();
+    rcl_test::sleep_for(std::chrono::duration_cast<std::chrono::nanoseconds>(trigger_period));
```

Comment:
`sleep_for` should not wake up spuriously: http://stackoverflow.com/questions/30538508/can-stdthis-threadsleep-for-have-spurious-wakeups

We had a problem with `std::condition_variable::wait_for` waking up spuriously, but that's documented behavior (that's why you have a predicate in condition variable).